### PR TITLE
Adding test coverage for Variable provider

### PR DESCRIPTION
**Pull request by ChrisUbben**
Could you both take a look at this for me? @Francisco-Gamino @JamesWTruher 
Thanks!

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        catch { $theError = $_ }
+        $theError.FullyQualifiedErrorId.StartsWith("NotSupported") | Should be $true
+    }
+}
+
+Describe "Validate special variables" -Tags "CI" {
+    It "Verify `$PSVersionTable.PSEdition" {
+        $PSVersionTable["PSEdition"] | Should Be "Core"
+    }
+}
```
Comment (Francisco-Gamino) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: On Github, you need to an extra empty line.

Comment (Francisco-Gamino) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Should we add more variables to the test? You can see all built-in variables if you run "dir env:"

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    AfterEach {
+        Remove-Variable -Name $testVarName -ErrorAction SilentlyContinue
+    }
+
+    It "Verify Negative New-Item" {
+        try { New-Item -Name $testVarName -Value 5 -Path "Variable:" -ErrorAction Stop } 
+        catch { $theError = $_ }
+        $theError.FullyQualifiedErrorId.StartsWith("Argument") | Should be $true
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Please follow the [error validation guidance for Pester tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md)

Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Fixed. Thanks Mike!

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 (hunk 1 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Added the new line.
We will analyze test coverage later and file a new issue with our results. At that point we will be adding additional test coverage.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        $existsAfter = Test-Path "Variable:${testVarName}"
+        $result = Get-Item "Variable:${testVarName}_Rename"
+        $existsBefore | Should Be $true
+        $existsAfter | Should Be $false
+        $result.Name | Should Be "${testVarName}_Rename"
+        $result.Value | Should Be $testVarValue
+    }
+}
+
+Describe "Validate basic negative test cases for Variable provider cmdlets" -Tags "CI" {
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Negative test cases should be tagged "Feature"

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    AfterEach {
+        Remove-Variable -Name $testVarName -ErrorAction SilentlyContinue
+    }
+
+    It "Verify Negative New-Item" {
+        try { New-Item -Name $testVarName -Value 5 -Path "Variable:" -ErrorAction Stop } 
+        catch { $theError = $_ }
+        $theError.FullyQualifiedErrorId | Should be "Argument,Microsoft.PowerShell.Commands.NewItemCommand"
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Your update for handling negative test cases did not get posted

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 (hunk 3 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Fixed. Thanks Mike!

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 (hunk 4 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: I changed
 $theError.FullyQualifiedErrorId.StartsWith("Argument") | Should be $true
to
$theError.FullyQualifiedErrorId | Should be "Argument,Microsoft.PowerShell.Commands.NewItemCommand"

Is this not what you were referring to when you said "Please follow the error validation guidance for Pester tests"?

Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: The guidance is this:
```
Try {
    New-Item -Name $testVarName -Value 5 -Path "Variable:" -ErrorAction Stop 
    throw "Expected exception not throw"
}
catch {
    $_.FullyQualifiedErrorId | Should be "Argument,Microsoft.PowerShell.Commands.NewItemCommand"
}
```

Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: I understand. I have fixed this.
Thanks Mike!

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    AfterEach {
+        Remove-Variable -Name $testVarName -ErrorAction SilentlyContinue
+    }
+
+    It "Verify Negative New-Item" {
+        try {
+            New-Item -Name $testVarName -Value 5 -Path "Variable:" -ErrorAction Stop 
+            throw "Expected exception not thrown"
+        }
+        catch { $theError = $_ }
```
Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: This still doesn't match the requested change. The "Should Be" assertion should be within the catch block

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1 (hunk 5 above)
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Variable.Tests.ps1]: Sorry about that. 
I have mad the changes now.

**Review by mirichmo — APPROVED** [review]

**Pull request merged**